src/MongoDB.Driver.Core/Core/Servers/Server.cs

```diff
@@ -314,11 +342,35 @@ private bool IsNotMaster(ServerErrorCode code, string message)
             return false;
         }
 
-        private bool IsNotMasterOrRecovering(ServerErrorCode code, string message)
+        private bool IsNotMasterErrorException(Exception exception)
+        {
+            return
+                exception is MongoCommandException commandException &&
+                IsNotMaster((ServerErrorCode)commandException.Code, commandException.ErrorMessage);
+        }
+
+        private bool IsStateChangeError(ServerErrorCode code, string message)
         {
             return IsNotMaster(code, message) || IsRecovering(code, message);
         }
 
+        private bool IsShutdownError(ServerErrorCode errorCode)
+        {
+            switch (errorCode)
+            {
+                case ServerErrorCode.InterruptedAtShutdown: // 1160
+                case ServerErrorCode.ShutdownInProgress: // 91
+                    return true;
+                default:
+                    return false;
+            }
+        }
```

Comment:
See [spec definition](https://github.com/mongodb/specifications/commit/7dd9c008be6edaf65368787d5e91c2ac3aeb95b8#diff-62b4da2e38e7d2a9c1a2d92d315895bbR1335)

Reply:
Or we can create a flag field as we already do for network errors. My point that these lines can/should be merged: https://github.com/vincentkam/mongo-csharp-driver/blob/41d027301c62ddcde96b88f22de24c2fe4448339/src/MongoDB.Driver.Core/Core/Servers/Server.cs#L325-L344 (only part of lines) and https://github.com/vincentkam/mongo-csharp-driver/blob/41d027301c62ddcde96b88f22de24c2fe4448339/src/MongoDB.Driver.Core/Core/Misc/ExceptionMapper.cs#L120-L157. But I guess this refactoring is out of the scope of this ticket, so I'm ok to do it later